src/Corax/IndexWriter.cs

```diff
@@ -361,23 +449,75 @@ public long Index(string id, Span<byte> data)
             return Index(idSlice, data);
         }
 
-        public unsafe long Index(string id, Span<byte> data, float documentBoost)
+        public long Index(string id, Span<byte> data, float documentBoost)
         {
             var entryId = Index(id, data);
             AppendDocumentBoost(entryId, documentBoost);
             return entryId;
         }
+        
+        private unsafe long Index(Slice id, Span<byte> data)
+        {
+            _numberOfModifications++;
+            Span<byte> buf = stackalloc byte[10];
+            var idLen = ZigZagEncoding.Encode(buf, id.Size);
+            int requiredSize = idLen + id.Size + data.Length;
+            // align to 16 bytes boundary to ensure that we have some (small) space for updating in-place entries
+            requiredSize += 16 - (requiredSize % 16);
+            var entryId = Container.Allocate(Transaction.LowLevelTransaction, _entriesContainerId, requiredSize, out var space);
+            buf.Slice(0, idLen).CopyTo(space);
+            space = space.Slice(idLen);
+            id.CopyTo(space);
+            space = space.Slice(id.Size);
+            data.CopyTo(space);
+            space = space.Slice(data.Length);
+            space.Clear();// clean any old data that may have already been there
+
```

Comment:
Add a comment here about the memory layout, I had to deal with that during my last bug on the null support and I could barely follow what was going on. Since we have changed it again, we need this to have a proper documentation in the code.

Reply:
I've not touched this part.